Objects/longobject.c

```diff
@@ -3292,17 +3216,20 @@ PyLong_AsDouble(PyObject *v)
 static Py_ssize_t
 long_compare(PyLongObject *a, PyLongObject *b)
 {
-    Py_ssize_t sign = Py_SIZE(a) - Py_SIZE(b);
+    if (_PyLong_BothAreSingleDigit(a, b)) {
+        return _PyLong_SingleDigitValue(a) - _PyLong_SingleDigitValue(b);
```

Comment:
This doesn't work well in the case where a is `LONG_MAX` and b is `-1`: this will not only result in `LONG_MIN`, this will also cause an overflow, which is undefined for signed integers.  Might want to consider using some other method, such as `(a > b) - (a < b)` or something like that.

Reply:
If the values fit in a "single digit", then they have at least one bit to spare, so that `+`, `-` cannot overflow.
I'll add comments to `_PyLong_IsSingleDigit` and `_PyLong_BothAreSingleDigit` to make that clear.